app/bundles/EmailBundle/Model/SendEmailToContact.php

```diff
@@ -399,21 +399,12 @@ protected function createContactStatEntry($email)
         ++$this->statBatchCounter;
 
         $stat = $this->mailer->createEmailStat(false, null, $this->listId);
-        // Store it in the saveEntities array so that every 20 are persisted to prevent mass duplciation resends if
-        // something goes wrong
-        $this->saveEntities[$email] = $stat;
+
         // Store it in the statEntities array so that the stat can be deleted if the transport fails the
         // send for whatever reason after flushing the queue
-        $this->statEntities[$email] = $stat;
+        $this->statHelper->storeStat($stat, $email);
```

Comment:
We are flushing each single Stats to the database here. This can have huge impact to the performance - we will be writing 20 times more often now. This is the purpose of this PR as I understand it, but do we really need it? Can we do just persist and flush it later?